libraries/AP_RTC/AP_RTC.cpp

```diff
@@ -84,12 +79,14 @@ void AP_RTC::set_utc_usec(uint64_t time_utc_usec, source_type type)
         // can't allow time to go backwards, ever
         return;
     }
-    WITH_SEMAPHORE(rsem);
 
-#if DEBUG_RTC_SHIFT
-    uint64_t old_utc = 0;
-    UNUSED_RESULT(get_utc_usec(old_utc));
-#endif
+#if AP_RTC_LOGGING_ENABLED
+    if (rtc_source_type != type) {
+        AP::logger().Write_RTC();
```

Comment:
If we got two SYSTEM_TIME messages (with different times) on the same flight, would this log both?
Not that I expect this to happen, but in my head it makes more sense to log whenever rtc_shift changes, instead of per source type